core/src/main/java/com/yahoo/oak/EntryHashSet.java

```diff
@@ -108,6 +109,16 @@ private int getKeyHash(int ei) {
         return HASH_CODEC.getFirst(getKeyHashAndUpdateCounter(ei));
     }
 
+    /**
+     * getUpdCntFromHash returns the key hash's update counter of the entry given by entry index "ei"
+     * (without key hash itself!).
+     */
+    private int getUpdCntFromHash(int ei) {
+        assert isIndexInBound(ei);
+        // extract the hash number (first) from the updates counter (second)
```

Comment:
I don't understand this comment: "extract the hash number (first)".
What is first here?

Reply:
Indeed some weird comment, probably old one. Rewritten.